modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -85,5 +85,32 @@ def report msg, response, error_only
       logger.error "Dnscmd failed:\n" + (response.is_a?(Array) ? response.join("\n") : "Response was not an array! #{response}")
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
+
+    def match_zone(fqdn)
+      dns_zones = ::Proxy::Dns::Plugin.settings.dns_zones || nil
+      found_match = false
+      weight = 0
+      matched_zone = fqdn.sub(/[^.]+./,'')
+
+      if dns_zones.is_a?(Array)
+        dns_zones.each do |zone|
+          zone_labels = zone.split(".").reverse
+          zone_weight = zone_labels.length
+          fqdn_labels = fqdn.split(".")
+          is_match = zone_labels.all? { |zone_label| zone_label == fqdn_labels.pop } || false
```

Comment:
no need for ```|| false``` -- #all? returns true if condition holds for elements of the array or false.